### PR TITLE
Add additional app service scm types

### PR DESCRIPTION
**Pull request by hbuckle**
Update scm_type validation with additional providers

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
hey @hbuckle 

Thanks for this PR :)

Whilst the code for this looks fine - would it be possible [to update the documentation for `app_service`](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/r/app_service.html.markdown#L152) to include these new values?

Thanks!

**Comment by hbuckle** [comment]
@tombuildsstuff - docs updated. I added VSTSRM as well, as that gets added by Azure Pipelines even though it isn't in the specs.

File: azurerm/helpers/azure/app_service.go [hunk 1]
```diff
@@ -159,8 +159,22 @@ func SchemaAppServiceSiteConfig() *schema.Schema {
...
+						string(web.ScmTypeExternalHg),
+						string(web.ScmTypeGitHub),
 						string(web.ScmTypeLocalGit),
+						string(web.ScmTypeNone),
+						string(web.ScmTypeOneDrive),
+						string(web.ScmTypeTfs),
+						string(web.ScmTypeVSO),
+						// Not in the specs, but is set by Azure Pipelines
+						// https://github.com/Microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureRmWebAppDeploymentV4/operations/AzureAppServiceUtility.ts#L19
+						"VSTSRM",
```
Comment (tombuildsstuff) [inline azurerm/helpers/azure/app_service.go]: @hbuckle heads up that I've opened an issue on the Swagger repository to get this added to the specs: https://github.com/Azure/azure-rest-api-specs/issues/5345

File: website/docs/r/app_service.html.markdown [hunk 2]
```diff
@@ -149,7 +149,21 @@ Additional examples of how to run Containers via the `azurerm_app_service` resou
...
+  * `Dropbox`
+  * `ExternalGit`
+  * `ExternalHg`
+  * `GitHub`
+  * `LocalGit`
+  * `None`
+  * `OneDrive`
+  * `Tfs`
+  * `VSO`
+  * `VSTSRM`
```
Comment (tombuildsstuff) [inline website/docs/r/app_service.html.markdown]: whilst this is a long list, we still tend to format this as:

```
`first`, `second`, `third` and `fourth`.
```

I'm going to push a commit to fix this (and also add it to the `app_service_slot` resource) - I hope you don't mind

File: azurerm/helpers/azure/app_service.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/helpers/azure/app_service.go]: it'd be good to add a TODO to that issue here ^

**Pull request merged**